.eslintrc.js

```diff
@@ -191,5 +162,42 @@ module.exports = {
         'react/prop-types': 'off',
       },
     },
+    {
+      files: ['*.tsx'],
+      rules: {
+        'react/prop-types': 'off',
+      },
+    },
+    {
+      files: ['*.spec.tsx', '*.spec.ts'],
+      rules: {
+        '@typescript-eslint/no-unused-expressions': 'off',
+        '@typescript-eslint/no-unused-vars': 'off',
+        '@typescript-eslint/no-use-before-define': 'off',
+        'import/prefer-default-export': 'off',
+        'jsx-a11y/anchor-has-content': 'off',
+        'jsx-a11y/anchor-is-valid': 'off',
+        'no-alert': 'off',
+        'no-console': 'off',
+        'no-empty-pattern': 'off',
+        'no-lone-blocks': 'off',
+        'no-shadow': 'off',
+        'react/default-props-match-prop-types': 'off',
+        'react/no-access-state-in-setstate': 'off',
+        'react/no-unused-prop-types': 'off',
+        'react/prefer-stateless-function': 'off',
+        'react/prop-types': 'off',
+        'react/require-default-props': 'off',
+        'react/state-in-constructor': 'off',
+        'react/static-property-placement': 'off',
+      },
+    },
+    {
+      files: ['packages/material-ui-icons/custom/**/*.js'],
```

Comment:
Why do we need to disable lint rules for JS files in this PR?

Reply:
I'm moving the rules from nested .eslintrc.js file to the global config

Reply:
Can we do this later?

Reply:
Later would probably mean never. What strategy do we want to adopt? for splitting the configuration. I like the nested .eslintrc because it means we can delete them with the source, I like the overrides because it means we can have a single view at whole the options, helping to simplify them. 

Revering

Reply:
Reverted